src/composables/transactionItemList.ts

```diff
@@ -0,0 +1,128 @@
+import { i18n } from '../store/plugins/languages';
+
+interface TransactionItemList extends IDefaultComposableOptions {
+  isMultisig?: boolean,
+  token: string,
+}
+
+type ITransactionElement = ITransaction | IActiveMultisigTx
+
+export const useTransactionList = ({
+  store,
+  isMultisig,
+  token,
+}: TransactionItemList) => {
+  const availableTokens = computed<ITokenList>(() => store.state.fungibleTokens.availableTokens);
+  const transactions = computed(() => store.state.transactions);
+  const getTxSymbol = computed(() => store.getters.getTxSymbol);
+  const getAccountPendingTransactions = computed<IActiveMultisigTx[]>(
+    () => store.getters.getAccountPendingTransactions,
+  );
+
+  const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
+
+  const {
+    searchPhrase,
+    displayMode,
+    FILTER_MODE,
+  } = useTransactionAndTokenFilter();
+
+  const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store });
+
+  function isFungibleTokenTx(tx: ITx) {
+    return Object.keys(availableTokens.value).includes(tx.contractId);
+  }
+
+  function filterProperTransactions(tr: ITransactionElement) {
+    const innerTx = getInnerTransaction(tr.tx!);
+    return !token || (
+      token !== AETERNITY_CONTRACT_ID
+        ? innerTx?.contractId === token
+        : (!innerTx.contractId || !isFungibleTokenTx(innerTx))
+    );
+  }
+
+  function filterTransactionByDisplayMode(tr: ITransactionElement) {
+    const { direction, isDex } = useTransactionTx({
+      store,
+      tx: tr.tx,
+      externalAddress: (tr as ITransaction).transactionOwner,
+    });
+
+    switch (displayMode.value.key) {
+      case FILTER_MODE.all:
+        return true;
+      case FILTER_MODE.dex:
+        return isDex.value;
+      case FILTER_MODE.out:
+        return direction.value === TX_FUNCTIONS.sent && !isDex.value;
+      case FILTER_MODE.in:
+        return direction.value === TX_FUNCTIONS.received;
+      default:
+        throw new Error(`${i18n.t('pages.recentTransactions.unknownMode')} ${displayMode.value.key}`);
+    }
+  }
+
+  function filterTransactionByPhrase(tr: ITransactionElement) {
+    return !searchPhrase.value || getTxSymbol.value(tr)
+      .toLocaleLowerCase()
+      .includes(searchPhrase.value.toLocaleLowerCase());
+  }
+
+  const loadedTransaction = computed((): ITransactionElement[] => {
+    const transactionList: ITransactionElement[] = [
+      ...getAccountPendingTransactions.value,
+      ...(transactions.value as any).loaded as ITransaction[],
+    ];
+
+    if (isMultisig && pendingMultisigTransaction.value?.tx) {
+      transactionList.push(pendingMultisigTransaction.value);
+    }
+
+    return transactionList;
+  });
+
+  const filteredTransactions = computed(
+    () => loadedTransaction.value
+      .filter(filterProperTransactions)
+      .filter(filterTransactionByDisplayMode)
+      .filter(filterTransactionByPhrase)
+      .sort(defaultTransactionSortingCallback),
```

Comment:
From my experience it's a bad pattern to create multiple callbacks for the filters and use them this way. For actions like this you can use the piping concept: https://www.freecodecamp.org/news/pipe-and-compose-in-javascript-5b04004ac937/

```
export function pipe<T = any[]>(fns: ((data: T) => T)[]) {
  return (data: T) => fns.reduce((currData, func) => func(currData), data);
}
```

Convert all the callback functions to do single filtering:
```
function filterProperTransactions(data: X) {
  return data.filter( ... );
}
```

And here use those 2 thing as follows:
```
() => pipe([
  filterProperTransactions,
  filterTransactionByDisplayMode,
  filterTransactionByPhrase,
  defaultTransactionSortingCallback,
])(loadedTransaction.value)
```